admin-api-server/src/analytics/service/analytics.service.ts

```diff
@@ -46,32 +49,72 @@ export class AnalyticsService {
   ): Promise<MetricEntity[]> {
     const qryRes = await this.#timeseries(params);
 
-    return qryRes.rows.map(
-      (row: any) =>
-        <MetricEntity>{
-          timestamp: Math.floor(row['timestamp'].getTime() / 1000),
-          value: Number(row['nft_count']),
-        },
-    );
+    const nftCountMap: Record<string, number> = {};
+    qryRes.rows.forEach((row: any) => {
+      const key = Math.floor(row['timestamp'].getTime() / 1000).toString();
+      nftCountMap[key] = Number(row['nft_count']);
+    });
+
+    return this.getTimestampData(nftCountMap, params.resolution);
   }
 
   async getTimeseriesSalesPriceVolume(
     params: MetricParams,
   ): Promise<MetricEntity[]> {
     const qryRes = await this.#timeseries(params);
 
-    return qryRes.rows.map(
-      (row: any) =>
-        <MetricEntity>{
-          timestamp: Math.floor(row['timestamp'].getTime() / 1000),
-          value: Number(
-            this.currencyService.convertToCurrency(
-              Number(row['price_volume']),
-              BASE_CURRENCY,
-            ),
-          ),
-        },
-    );
+    const priceVolumeMap: Record<string, number> = {};
+    qryRes.rows.forEach((row: any) => {
+      const key = Math.floor(row['timestamp'].getTime() / 1000).toString();
+
+      priceVolumeMap[key] = Number(
+        this.currencyService.convertToCurrency(
+          Number(row['price_volume']),
+          BASE_CURRENCY,
+        ),
+      );
+    });
+
+    return this.getTimestampData(priceVolumeMap, params.resolution);
+  }
+
+  getTimestampData(
+    timestampMap: Record<string, number>,
+    resolution: Resolution,
+  ): MetricEntity[] {
+    let manipulate: ManipulateType;
+    switch (resolution) {
+      case Resolution.Hour:
+        manipulate = Resolution.Hour as ManipulateType;
+        break;
+      case Resolution.Day:
+        manipulate = Resolution.Day as ManipulateType;
+        break;
+      case Resolution.Week:
+        manipulate = Resolution.Week as ManipulateType;
+        break;
+      case Resolution.Month:
+        manipulate = Resolution.Month as ManipulateType;
```

Comment:
not quite sure how this works (how this `as ManipulateType` works here, is it just coincidence the resolution names are the same?), but if this works it'd be a nice small improvement I think:

```
...

case Resolution.Hour:
case Resolution.Day:
case Resolution.Week:
case Resolution.Month:
  manipulate = resolution as ManipulateType;
  break;

...
```

Reply:
Yes it is coincidence

Reply:
actually.. then i am more for your original approach but with explicit mapping. ie:

```
....

case Resolution.Hour:
  manipulate = ManipulateType.Hour;
  break;

...
```

Reply:
Done without type casting. I am confused though because I am sure I tried it without it already once and it didn't work. Gave me no errors now.